core/src/main/java/com/datastrato/gravitino/utils/ClientPoolImpl.java

```diff
@@ -89,6 +89,12 @@ protected boolean isConnectionException(Exception exc) {
 
   @Override
   public void close() {
```

Comment:
Is it possible close will be invoked twice?

Reply:
I close it manually by `clientPoolCache.asMap().forEach((key, value) -> value.close());` and the removal listener will close again if time expires